consoleApp/src/main/java/by/issoft/consoleApp/handlers/sort/SortCommand.java

```diff
@@ -11,28 +10,27 @@
 import java.io.*;
 import java.net.URISyntaxException;
 import java.net.URL;
+import java.util.Collections;
 import java.util.Comparator;
-import java.util.List;
+import java.util.Map;
+
+public class SortCommand implements AppCommand {
+    private final static XmlMapper XML_MAPPER = new XmlMapper();
 
-public class SortHandler implements MyHandler {
     @Override
     public void execute(Store store) {
-        Comparator<Product> resultComparator = getProductComparator();
+        Map<String, String> productSortConfig = getProductSortConfig();
+        Comparator<Product> resultComparator = getProductComparator(productSortConfig);
 
-        store.getCategories()
-                .stream()
-                .map(Category::getProducts)
-                .flatMap(List::stream)
-                .sorted(resultComparator)
+        prepareSortProductList(store, resultComparator)
                 .forEach(System.out::println);
     }
 
-    private Comparator<Product> getProductComparator() {
-        ProductSortConfig value = getProductSortConfig();
+    private Comparator<Product> getProductComparator(Map<String, String> productSortConfig) {
 
-        boolean sortProductNameAsc = "asc".equals(value.getNameSortConfig());
-        boolean sortProductPriceAsc = "asc".equals(value.getPriceSortConfig());
-        boolean sortProductRateAsc = "asc".equals(value.getRateSortConfig());
+        boolean sortProductNameAsc = "asc".equals(productSortConfig.get("name"));
```

Comment:
please support dynamic number of attributes of Product class (e/g just name)

Reply:
Done